src/components/LocationAutocompleteField/LocationAutocompleteField.tsx

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+/* eslint-disable spaced-comment */
+import { FieldProps } from 'formik';
+import LocationAutocomplete from 'components/LocationAutocomplete';
+import { FormValuesType } from 'components/HistoricalForecast/validation';
+import Location from 'types/location';
+
+interface Props extends FieldProps<Location, FormValuesType> {
+  id: string;
+}
+
+export default function LocationAutoCompleteField({
+  field: { value },
+  form: { setValues },
+  meta,
+  ...props
+}: Props) {
```

Comment:
Be careful here: you need to write all location data not at the top level of values but in `location` field
In order to see a problem, just log `values` on location selection

Also you need to initialize `location` field as null in order to surpress an error message about transition from uncontrolled to controlled component